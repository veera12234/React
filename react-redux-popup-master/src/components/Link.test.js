import React from 'react'
import PropTypes from 'prop-types'
//React Render Jest
import renderer from 'react-test-renderer';
import Link from './Link.js'

describe('Link Component', () => {

    test('Active Page - No Link', () => {
        const LinkViewComponent = renderer
            .create(<Link active={false} onClick={Function}>Test</Link>);

        let LinkView = LinkViewComponent.toJSON();
        expect(LinkView).toMatchSnapshot();

        LinkView.props.onClick({
            preventDefault: () => {
            }
        });

        LinkView = LinkViewComponent.toJSON();
        expect(LinkView).toMatchSnapshot();

    });

    test('InActive Page - Show Link', () => {
        const LinkViewComponent = renderer
            .create(<Link active={true} onClick={Function}>Test</Link>);

        let LinkView = LinkViewComponent.toJSON();
        expect(LinkView).toMatchSnapshot();
    });

});
