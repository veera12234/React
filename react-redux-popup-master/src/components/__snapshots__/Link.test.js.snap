// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Component Active Page - No Link 1`] = `
<a
  href="#"
  onClick={[Function]}
>
  Test
</a>
`;

exports[`Link Component Active Page - No Link 2`] = `
<a
  href="#"
  onClick={[Function]}
>
  Test
</a>
`;

exports[`Link Component InActive Page - Show Link 1`] = `
<span>
  Test
</span>
`;
